src/gd_kotlin.cpp

```diff
@@ -480,6 +521,43 @@ void GDKotlin::check_and_copy_jar(const String& jar_name) {
 #endif
 }
 
+jni::JObject GDKotlin::_prepare_class_loader(jni::Env& p_env, jni::Jvm::Type type) {
+    if (type == jni::Jvm::GRAAL) {
+        return jni::JObject();
+    }
+#ifdef __ANDROID__
+    String bootstrap_jar_file{"godot-bootstrap-dex.jar"};
+    String main_jar_file{"main-dex.jar"};
+#else
+    String bootstrap_jar_file{"godot-bootstrap.jar"};
+#endif
+
+    _check_and_copy_jar(bootstrap_jar_file);
+
+#ifdef TOOLS_ENABLED
+    String bootstrap_jar{OS::get_singleton()->get_executable_path().get_base_dir() + "/godot-bootstrap.jar"};
+#else
+    String bootstrap_jar{ProjectSettings::get_singleton()->globalize_path(vformat("user://%s", bootstrap_jar_file))};
+#endif
+
+#ifdef TOOLS_ENABLED
+    JVM_CRASH_COND_MSG(!FileAccess::exists(bootstrap_jar),
+                       "No godot-bootstrap.jar found! This file needs to stay alongside the godot editor executable!")
+#elif DEBUG_ENABLED
+    JVM_CRASH_COND_MSG(!FileAccess::exists(bootstrap_jar), "No godot-bootstrap.jar found!")
+#endif
+
+    LOG_INFO(vformat("Loading bootstrap jar: %s", bootstrap_jar))
+
+    jni::JObject class_loader {ClassLoader::provide_loader(p_env, bootstrap_jar, jni::JObject(nullptr))};
+    ClassLoader::set_default_loader(class_loader);
+    class_loader.delete_local_ref(p_env);
+
+    class_loader = ClassLoader::get_default_loader();
+
+    return class_loader;
+}
```

Comment:
Shouldn't we move this into a separate class?

Reply:
We need a check and copy jar. This file is ugly and was already before this PR. This PR just makes this more visible.
We call other methods from `GDKotlin` in this method.
IMO GDKotlin needs a complete refactor in a seperate PR.

Reply:
Agreed.

Reply:
I did make a follow up issue: https://github.com/utopia-rise/godot-kotlin-jvm/issues/258